scapy/arch/windows/__init__.py

```diff
@@ -67,16 +69,66 @@ def _encapsulate_admin(cmd):
     # To get admin access, we start a new powershell instance with admin rights, which will execute the command
     return "Start-Process PowerShell -windowstyle hidden -Wait -Verb RunAs -ArgumentList '-command &{%s}'" % cmd
 
+class _PowershellManager(Thread):
+    """Used only when scapy boots. Instance used to send multiple
+    commands on the same Powershell process. Skips the powershell
+    booting time to speed up scapy's boot"""
+    def __init__(self):
+        self.process = sp.Popen([conf.prog.powershell, "-NoLogo", "-NonInteractive", "-Command", "-"], # Start & redirect input
+                         stdout=sp.PIPE,
+                         stdin=sp.PIPE,
+                         stderr=sp.STDOUT,
+                         universal_newlines=True)
+        self.buffer = []
+        self.running = True
+        self.event = Event()
+        Thread.__init__(self)
+        self.start()
+
+    def run(self):
+        while self.running:
+            read_line = self.process.stdout.readline()
+            if read_line == "scapy_end\n":
+                self.event.set()
+            else:
+                self.buffer.append(read_line)
+    def query(self, command, cmd=False):
+        if cmd:
```

Comment:
cmd is needed only when not running - please move under if

Reply:
Done. Nice catch !